src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -220,12 +218,18 @@ protected function prepareJsonResponse($request, Exception $e)
 
         $headers = $this->isHttpException($e) ? $e->getHeaders() : [];
 
-        return response(json_encode([
-            'message' => $e->getMessage(),
-            'file' => $e->getFile(),
-            'line' => $e->getLine(),
-            'trace' => $e->getTrace(),
-        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES), $status, array_merge($headers, [
+        if (config('app.debug')) {
+            $response = [
+                'message' => $e->getMessage(),
+                'file' => $e->getFile(),
+                'line' => $e->getLine(),
+                'trace' => $e->getTrace(),
```

Comment:
Maybe, but it is relate these code, about the exception. It doesn't matter how the code work. Please consider the exception code, the translation. The file name and the exception line shown in production may not be safe for the server.